docs/examples/metadata_update/producer_metadata_disconnection_with_backoff.py

```diff
@@ -0,0 +1,106 @@
+import asyncio
+import time
+
+from rstream import (
+    AMQPMessage,
+    OnClosedErrorInfo,
+    RouteType,
+    SuperStreamProducer,
+)
+
+SUPER_STREAM = "invoices"
+MESSAGES = 100000000
+producer_closed = False
+
+
+# this value will be hashed using mumh3 hashing algorithm to decide the partition resolution for the message
+async def routing_extractor(message: AMQPMessage) -> str:
+    return message.application_properties["id"]
+
+
+async def publish():
+    async def on_metadata_update(on_closed_info: OnClosedErrorInfo) -> None:
+
+        if on_closed_info.reason == "MetaData Update":
+            print(
+                "metadata changed for stream : "
+                + str(on_closed_info.streams[0])
+                + " with code: "
+                + on_closed_info.reason
+            )
+            await asyncio.sleep(2)
+            # reconnect just if the partition exists
+            if await super_stream_producer.stream_exists((on_closed_info.streams[0])):
+                await super_stream_producer.reconnect_stream(on_closed_info.streams[0])
+
+        else:
+            print(
+                "connection has been closed from stream: "
+                + str(on_closed_info.streams)
+                + " for reason: "
+                + str(on_closed_info.reason)
+            )
+
+            await asyncio.sleep(2)
+            # reconnect just if the partition exists
+            for stream in on_closed_info.streams:
```

Comment:
I would add that if the error code is:
`streamDoesNotExist` just close the producer/consumer